pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -77,135 +44,56 @@ func (current *Resources) Remove(incoming Resources, all bool) {
-func (current *Resources) Fits(incoming Resources) error {
-	exceededResources := []string{}
-
-	// Define function for checking int resources to keep code DRY
-	checkResource := func(resource string, currentVal, incomingVal int) {
-		if currentVal != Unlimited && currentVal < incomingVal {
-			exceededResources = append(exceededResources, resource)
-		}
+func SubQuantity(c, i resource.Quantity) resource.Quantity {
+	if c.Equal(comparableUnlimitedQuntity) || i.Equal(comparableUnlimitedQuntity) {
+		return c
 	}
-
-	// Define function for checking quantity resources to keep code DRY
-	checkQuantityResource := func(resource string, currentVal, incomingVal resource.Quantity) {
-		if !currentVal.Equal(UnlimitedQuantity()) && currentVal.Cmp(incomingVal) < 0 {
-			exceededResources = append(exceededResources, resource)
-		}
+	c.Sub(i)
+	if c.CmpInt64(0) < 0 {
+		c.Set(0)
 	}
+	return c
+}
 
-	checkResource("Apps", current.Apps, incoming.Apps)
-	checkResource("Containers", current.Containers, incoming.Containers)
-	checkResource("Jobs", current.Jobs, incoming.Jobs)
-	checkResource("Volumes", current.Volumes, incoming.Volumes)
-	checkResource("Secrets", current.Secrets, incoming.Secrets)
-	checkResource("Images", current.Images, incoming.Images)
-	checkResource("Projects", current.Projects, incoming.Projects)
-
-	checkQuantityResource("VolumeStorage", current.VolumeStorage, incoming.VolumeStorage)
-	checkQuantityResource("Memory", current.Memory, incoming.Memory)
-	checkQuantityResource("Cpu", current.CPU, incoming.CPU)
-
-	// Build an aggregated error message for the exceeded resources
-	if len(exceededResources) > 0 {
-		return fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", "))
+func Fits(toAppend []string, resource string, current, incoming int) []string {
+	if current != Unlimited && current < incoming {
+		return append(toAppend, resource)
 	}
-
-	return nil
+	return toAppend
 }
 
-// NonEmptyString will return a string representation of the non-empty
-// Resources within the struct.
-func (current *Resources) NonEmptyString() string {
-	var resources []string
-
-	// Define function for checking int resources to keep code DRY
-	checkResource := func(resource string, value int) {
-		if value > 0 {
-			resources = append(resources, resource)
-		}
+func FitsQuantity(toAppend []string, resource string, current, incoming resource.Quantity) []string {
+	if !current.Equal(comparableUnlimitedQuntity) && current.Cmp(incoming) < 0 {
+		return append(toAppend, resource)
 	}
+	return toAppend
+}
 
-	// Define function for checking quantity resources to keep code DRY
-	checkQuantityResource := func(resource string, currentVal resource.Quantity) {
-		if !currentVal.IsZero() {
-			resources = append(resources, resource)
-		}
+// ResourceToString will return a string representation of the resource and value
+// if its the value is greater than 0.
+func ResourceToString(resource string, value int) []string {
```

Comment:
Why does this return `[]string` instead of `string`?

Reply:
I implemented it this way for a reason that no longer makes sense to me. I'll simplify this down and it'll use string instead.